spec/services/proofing/aamva/proofer_spec.rb

```diff
@@ -1,6 +1,117 @@
 require 'rails_helper'
 require 'ostruct'
 
+# This helper method generates a set of tests for a specific attribute.
+# They assume that `verification_response` contains a DLDV XML response
+# where all attributes are present and valid, and `result` contains
+# the result from the proofer.
+# @param [Symbol] attribute_name
+# @param [String] match_indicator_name Tag name for the match indicator in the DLDV XML response
+# @param [Boolean] required Whether this attribute must verfy for overall success
+# @param [nil,Symbol] required_part_of The meta-attribute this attribute is required for
+#                                      (e.g. :address for :city)
+# @param [nil,Symbol] optional_part_of The meta-attribute this attribute is an optional part of
+#                                      (e.g. :address for :address2)
+def test_aamva_attribute(
+  attribute_name,
+  match_indicator_name:,
+  required: false,
+  required_part_of: nil,
+  optional_part_of: nil
+)
+  context 'when unverified' do
+    let(:verification_response) do
+      XmlHelper.modify_xml_at_xpath(
+        super(),
+        "//#{match_indicator_name}",
+        'false',
+      )
+    end
+
+    if required
+      it('makes the result not succeed') do
+        expect(result.success?).to be false
+      end
+    else
+      it 'does not stop result from succeeding' do
+        expect(result.success?).to be true
+      end
+    end
```

Comment:
so my thinking with defining the separate methods in the other comment (https://github.com/18F/identity-idp/pull/11282#discussion_r1775298406) is that with like 3 options, it gets a little tricky to follow this branching. I know it's in the spirit of "don't repeat yourself" but I think that this can be tricky to refactor in the future if some new unexpected kind of attribute comes along

If this isn't clear, LMK and I can make a patch that explains what I'm talking about

Reply:
@zachmargolis wdyt of [this style](https://github.com/18F/identity-idp/blob/3c9f96a8f13f1a7db491b5b6fe336fd1491be0d3/spec/services/proofing/aamva/proofer_spec.rb#L143-L158)

Reply:
that is much fancier than I could have ever imagined, I think it's much clearer than all the branching we had before! nice work!